src/Microsoft.Health.Dicom.Core/Features/Delete/DeleteService.cs

```diff
@@ -92,34 +92,43 @@ public async Task<(bool Success, int RetrievedInstanceCount)> CleanupDeletedInst
             try
             {
                 var deletedInstanceIdentifiers = (await _indexDataStore
-                    .RetrieveDeletedInstancesAsync(
+                    .RetrieveDeletedInstancesWithPropertiesAsync(
```

Comment:
Won't this fail when code is updated and schema is not updated yet?

In SqlIndexDataStoreV1.cs you will have to call the old procedure and populate the OriginalWatermark as null to make this work seemlessly

Reply:
I thought I called it, similar to what you did instanceWithProperties. Totally missed it, thanks for catching it

Reply:
Done